scanner/src/main/kotlin/utils/FileListResolver.kt

```diff
@@ -43,8 +44,11 @@ internal class FileListResolver(
     fun resolve(provenance: KnownProvenance): FileList {
         storage.getFileList(provenance)?.let { return it }
 
-        return provenanceDownloader.download(provenance).let { dir ->
-            createFileList(dir).also { storage.putFileList(provenance, it) }
+        val dir = provenanceDownloader.download(provenance)
+
+        return createFileList(dir).also {
+            storage.putFileList(provenance, it)
```

Comment:
I believe `putFileList()` could throw an exception.
So, the deletion should go into some `finally` block.